osu.Game/Overlays/KeyBinding/KeyBindingsSubsection.cs

```diff
@@ -32,20 +33,25 @@ protected KeyBindingsSubsection(int? variant)
         }
 
         [BackgroundDependencyLoader]
-        private void load(KeyBindingStore store)
+        private void load(RealmContextFactory realmFactory)
         {
-            var bindings = store.Query(Ruleset?.ID, variant);
+            var rulesetId = Ruleset?.ID;
 
-            foreach (var defaultGroup in Defaults.GroupBy(d => d.Action))
+            using (var realm = realmFactory.GetForRead())
```

Comment:
not sure about the scoping of this `using` - it appears that disposing a `Realm` frees some system-level handles so it might be good to drop them as soon as we're done fetching what we want even if there is no limit on concurrency?

also, I assume `.GetForRead()` on the factory is for non-update threads, and `.Context` is for update thread, right? is that what you were planning to expose framework's `ThreadSafety` for?

Reply:
Originally I was attempting this without detaching, which is why the whole block is within the context. I think it could be changed but I doubt it will have any discernible difference in this particular usage.

> also, I assume .GetForRead() on the factory is for non-update threads, and .Context is for update thread, right? is that what you were planning to expose framework's ThreadSafety for?

That's correct, yes.